src/Sylius/Bundle/CoreBundle/spec/EventListener/ProductDeletionListenerSpec.php

```diff
@@ -25,62 +25,59 @@ final class ProductDeletionListenerSpec extends ObjectBehavior
 {
     function let(
         RequestStack $requestStack,
-        ProductAwareRuleUpdaterInterface $firstUpdater,
-        ProductAwareRuleUpdaterInterface $secondUpdater,
+        ProductPromotionRuleCheckerInterface $productPromotionRuleChecker,
     ): void {
-        $this->beConstructedWith($requestStack, $firstUpdater, $secondUpdater);
+        $this->beConstructedWith($requestStack, $productPromotionRuleChecker);
     }
 
     function it_throws_an_exception_when_subject_is_not_a_product(GenericEvent $event): void
     {
         $event->getSubject()->willReturn('subject');
 
-        $this->shouldThrow(\InvalidArgumentException::class)->during('removeProductFromPromotionRules', [$event]);
+        $this->shouldThrow(\InvalidArgumentException::class)->during('protectFromRemovingProductThatIsAssignedToPromotionRule', [$event]);
     }
 
-    function it_does_nothing_when_rule_updaters_do_not_return_any_codes(
+    function it_does_nothing_when_product_is_not_assigned_to_rule(
         RequestStack $requestStack,
-        ProductAwareRuleUpdaterInterface $firstUpdater,
-        ProductAwareRuleUpdaterInterface $secondUpdater,
+        ProductPromotionRuleCheckerInterface $productPromotionRuleChecker,
         SessionInterface $session,
         GenericEvent $event,
         ProductInterface $product,
     ): void {
         $event->getSubject()->willReturn($product);
 
-        $firstUpdater->updateAfterProductDeletion($product)->willReturn([]);
-        $secondUpdater->updateAfterProductDeletion($product)->willReturn([]);
+        $productPromotionRuleChecker->hasProduct($product)->willReturn(false);
 
         $requestStack->getSession()->willReturn($session);
         $session->getBag('flashes')->shouldNotBeCalled();
```

Comment:
```suggestion
        $requestStack->getSession()->shouldNotBeCalled();
```
The session mock is not needed